packages/types/src/user/user.ts

```diff
@@ -45,3 +51,14 @@ export function getDisplayName(user: User): string {
 
   return user.email
 }
+
+export function hasFlag<T extends { flags: UserFlag[] } | { userFlags: UserFlag[] }>(
+  userResolvable: T,
+  flag: UserFlag
+): boolean {
+  if ("flags" in userResolvable) {
+    return userResolvable.flags.includes(flag)
+  }
+
+  return userResolvable.userFlags.includes(flag)
+}
```

Comment:
Maybe we should just do `getUserFlags()` and check at callsite? Also under what circumstances is the field name different? This should be consistent.
```suggestion
export function getUserFlags(user: User & { flags: UserFlag[] }) {
  return new Set(user.flags)
}
```